src/main/java/dev/ecattez/shahmat/board/PieceBox.java

```diff
@@ -0,0 +1,42 @@
+package dev.ecattez.shahmat.board;
+
+import dev.ecattez.shahmat.board.pawn.Pawn;
+
+public class PieceBox implements PieceFactory {
+
+    @Override
+    public Piece createPiece(PieceType type, PieceColor color) {
+        return type.accept(new PieceTypeVisitor<Piece>() {
+            @Override
+            public Piece visitPawn() {
+                return new Pawn(color);
+            }
+
+            @Override
+            public Piece visitRook() {
+                return null;
```

Comment:
Same, `throw new NotImplemented` ?

Reply:
If it throw an exception, tests about promotion breaks. Maybe all piece should be available.